lib/std/Build/Step/Compile.zig

```diff
@@ -1771,12 +1764,12 @@ fn getZigArgs(compile: *Compile, fuzz: bool) ![][]const u8 {
             for (arg, 0..) |c, arg_idx| {
                 if (c == '\\' or c == '"') {
                     // Slow path for arguments that need to be escaped. We'll need to allocate and copy
-                    var escaped = try ArrayList(u8).initCapacity(arena, arg.len + 1);
-                    const writer = escaped.writer();
-                    try writer.writeAll(arg[0..arg_idx]);
+                    var escaped: std.ArrayListUnmanaged(u8) = .empty;
+                    try escaped.ensureTotalCapacityPrecise(arena, arg.len + 1);
+                    try escaped.appendSlice(arena, arg[0..arg_idx]);
```

Comment:
To avoid wasting memory with an arena:
```suggestion
                    var escaped: std.ArrayListUnmanaged(u8) = .empty;
                    try escaped.ensureTotalCapacityPrecise(arena, arg.len + blk: {
                        var num_to_escape: usize = 0;
                        for (arg) |c2|
                            num_to_escape += @intFromBool(c2 == '\\' or c2 == '"');
                        break :blk num_to_escape;
                    });
                    escaped.appendSliceAssumeCapacity(arg[0..arg_idx]);
```
#24104 would make this look neater.